src/include/mpir_coll.h

```diff
@@ -101,6 +101,8 @@ int MPIR_Allreduce_intra_recursive_doubling(const void *sendbuf, void *recvbuf,
                                       MPI_Datatype datatype, MPI_Op op, MPIR_Comm *comm_ptr, MPIR_Errflag_t *errflag);
 int MPIR_Allreduce_intra_reduce_scatter_allgather(const void *sendbuf, void *recvbuf, int count,
                                             MPI_Datatype datatype, MPI_Op op, MPIR_Comm *comm_ptr, MPIR_Errflag_t *errflag);
+int MPIR_Allreduce_intra_reduce_scatter_allgather_optimal(const void *sendbuf, void *recvbuf, int count,
```

Comment:
We should make the naming consistent with the 2 added functions. This commit has:
```
MPIR_Allreduce_intra_reduce_scatter_allgather_optimal
MPIR_Reduce_intra_redscat_gather_optimal
```
I would prefer:
```
MPIR_Allreduce_intra_redscat_allgather_optimal
MPIR_Reduce_intra_redscat_gather_optimal
```

Reply:
Note that we replaced `redscat` with `reduce_scatter` everywhere to keep it consistent.  Please don't make it inconsistent again.

Reply:
Also, `_optimal` is very misleading.  It's not optimal with respect to performance.  It's reducing memory space to a constant value, and it's unclear if even that is optimal.  Unless we theoretically proved that there does not exist a smaller constant value in which this can be implemented, I'd stay away from the word `optimal`.

Reply:
Noted the comments and will update accordingly